lib/p2p/Pool.ts

```diff
@@ -68,6 +68,20 @@ class Pool extends EventEmitter {
     return peer;
   }
 
+  public removeOutbound = async (address: string, port: number): Promise<void> => {
+    const socketAddress = new SocketAddress(address, port);
+    if (this.peers.has(socketAddress)) {
+      const peer = Peer.fromOutbound(socketAddress);
+      this.peers.remove(peer);
+      peer.destroy();
+      this.hosts.removeHost(socketAddress);
+    } else {
+      const err = errors.NOT_CONNECTED(address.toString());
+      this.logger.info(err.message);
+      throw err;
+    }
+  }
+
```

Comment:
* if the peer exists, you need to destroy him and to make sure he’s removed from the peers list (you might even just want to delegate the destroying responsibility to the peers list). At the moment you’re checking if he exists, and if so, you’re creating a new peer, and then remove the old peer according to the new peer `socketAddress`. Even if it can somehow work, it need to be fixed.
* you don’t need to remove the host, which is saved for future connection. it might be a good idea to have an option for it, but it shouldn’t be the only supported behavior.
* the disconnection implementation shouldn't be under `removeOutbound`. Although when initiating a new connection it will always be an outbound, when disconnecting this differentiation is not necessary. You'll want to support disconnection also if it's an inbound connection, and anyway there shouldn't be any difference in the implementation of the two.